book/security.rst

```diff
@@ -0,0 +1,660 @@
+a controller method. Restricting access can be done in two different ways:
+
+#. Use the ``@Acl`` annotation on a controller method, providing the entity class name and the
+   permission to check for:
+
+   .. code-block:: php-annotations
+
+       // src/Acme/DemoBundle/Controller/ProductController.php
+       namespace Acme\DemoBundle\Controller;
+
+       use Oro\Bundle\SecurityBundle\Annotation\Acl;
+       use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+       class ProductController extends Controller
+       {
+           /**
+            * @Acl(
+            *   id="product_edit",
+            *   type="entity",
+            *   class="AcmeDemoBundle:Product",
+            *   permission="EDIT"
+            * )
+            */
+           public function editAction()
+           {
+               // ...
+           }
+       }
+
+#. When you are in the need to perform a particular check repeatedly, writing the ``@Acl`` over and
+   over again becomes a tedious task. This becomes even more a serious issue when your requirements
+   change and you have to change a lot of ACLs. Luckily, you can configure an ACL globally in your
+   bundle configuration and refer to using the ACL id using the ``@AclAncestor`` annotation.
+
+   The ACL configuration from the example above looks like this:
+
+   .. code-block:: yaml
+
+       # src/Acme/DemoBundle/Resources/config/acl.yml
+       product_edit:
+           type: entity
+           class: AcmeDemoBundle:Product
+           permission: EDIT
+
+   The annotation of your controller method becomes a lot smaller then:
+
+   .. code-block:: php-annotations
+
+       // src/Acme/DemoBundle/Controller/ProductController.php
+       namespace Acme\DemoBundle\Controller;
+
+       use Oro\Bundle\SecurityBundle\Annotation\AclAncestor;
+       use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+       class ProductController extends Controller
+       {
+           /**
+            * @AclAncestor("product_edit")
+            */
+           public function editAction()
+           {
+               // ...
+           }
+       }
+
+   Sometimes you want to protect a controller method coming from code that you do not control.
+   Therefore, you cannot add the ``@AclAncestor`` annotation to it. Use the ``bindings`` key in the
+   YAML configuration of your ACL to define which method(s) should be protected:
+
+   .. code-block:: yaml
+
+       # src/Acme/DemoBundle/Resources/config/acl.yml
+       product_edit:
+           type: entity
+           class: AcmeDemoBundle:Product
+           permission: EDIT
+           bindings:
+               - class: Acme\DemoBundle\Controller\ProductController
+                 method: editAction
+
+Using Param Converters
+~~~~~~~~~~~~~~~~~~~~~~
+
+When using the `@ParamConverter annotation`_ from the SensioFrameworkExtraBundle, the behavior is a
+bit different. Besides just checking the ``@Acl`` annotations, it is also checked if the Doctrine
```

Comment:
it is not different. In this case we can check entity object by access level. 

For example, we have action without param converter rules but with ACL annotation. In this case we can check permission only on 'class' level. What this means. In this case if you have access level > than 'none' , you will grant access. And only if you have 'none' access level, access will be denied.

it is not different. In this case we can check entity object by access level. 

For example, we have action without param converter rules but with ACL annotation. In this case we can check permission only on 'class' level. What this means. In this case if you have access level > than 'none' , you will grant access. And only if you have 'none' access level, access will be denied.

For another example, if action have ACL annotation that should check 'VIEW' permission on 'AcmeDemoBundle:TestEntity' and ParamConverter rules, wich will convert input 'id' parameter into 'AcmeDemoBundle:TestEntity' Doctrine entity object. In this case will be checked access by 'object' and we able to check access level to this entity record. We does not check Doctrine query here.